utils.py

```diff
@@ -0,0 +1,217 @@
+        self.killed = True
+
+    def addToTable(self, entry):
+        if self.mainTable is None:
+            self.mainTable = TableView()
+        self.mainTable.addRow(entry)
+
+    def loadGraph(self, plots_in):
+        self.plots = plots_in
+        if self.plots:
+            self.mainFigure = Figure(self.plots)
+
+    def linePlot(self, mat, label, startPoint, endPoint, resolution=100, interpolation="bilinear"):
+
+        direcion = endPoint - startPoint
+        distance = np.linalg.norm(direcion)
+        distances = []
+        values = []
+
+        for i in range(resolution):
+
+            point = startPoint + i * direcion / resolution
+
+            if interpolation == "bilinear":
+                x = int(point[0])
+                y = int(point[1])
+                dx = point[0] - x
+                dy = point[1] - y
+                val = (np.array([[1 - dx, dx]]) @ mat[x : x + 2, y : y + 2] @ np.array([[1 - dy], [dy]]))[0, 0]
+
+            if interpolation == "nearest_neighbour":
+                val = mat[round(point[0]), round(point[1])]
+
+            values.append(val)
+            distances.append(i * distance / resolution)
+
+        self.addToTable([label, min(values), max(values), sum(values) / len(values)])
+
+        if self.mainFigure:
+            self.mainFigure.killFigure()
+        self.plots.append([distances, values, label])
+        self.mainFigure = Figure(self.plots)
+
+
+def saveImage(window):
+    imageSurface = window.imsurf
+    overlays = window.overlays
+    exthandler = window.exthandler
+    Tk().withdraw()
+    file = filedialog.asksaveasfile()
+
+    if file:
+
+        imgdata = pygame.surfarray.pixels3d(imageSurface).astype(np.uint8)
+        imgdata = np.swapaxes(imgdata, 0, 1)
+
+        filename = file.name
+        try:
+            Image.fromarray(imgdata).save(filename)
+        except ValueError:
+            filename += ".png"
+            Image.fromarray(imgdata).save(filename)
+
+        if messagebox.askquestion("Save options", "Do you want to save with the lines?") == "yes":
+            imageSurface.blit(overlays, (0, 0))
+
+            data = SaveData()
+            data.plots = window.exthandler.plots
+            data.tableEntries = []
+            data.mat = window.mat
+            data.mat_orig = window.mat_orig
+            data.mat_emm = window.mat_emm
+            data.raw = window.raw
+            data.meta = window.meta
+            data.overlays = pygame.image.tostring(overlays, "RGBA")
+
+            if exthandler.mainFigure:
+                exthandler.mainFigure.saveFig(".".join(filename.split(".")[:-1]) + "_plot.png")
+            if exthandler.mainTable:
+                exthandler.mainTable.writeToFile(".".join(filename.split(".")[:-1]) + "_values.csv")
+                data.tableEntries = exthandler.mainTable.data
+
+            with open(".".join(filename.split(".")[:-1]) + ".pkl", "wb") as f:
+                pickle.dump(data, f, -1)
+
+
+def openImage():
```

Comment:
When the program starts, it directly loads an image selection window. For someone who doesn't know what they're looking at, this could be confusing. Can you add a title like "Open Thermal Image" or something like that?
![image](https://user-images.githubusercontent.com/10851575/126545615-f85d764b-2134-4b7b-970a-62e9b1b35dda.png)